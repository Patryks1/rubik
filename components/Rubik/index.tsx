import React, { FC, useState } from 'react';
import { RUBIK_INITIAL_STATE } from '../../constants';
import { Color, FaceDirection, IRubikFace } from '../../interfaces';
import Face from './Face';
import Controls from './Controls';

const Rubik: FC = () => {
  const [cube, setCube] = useState<IRubikFace[]>(RUBIK_INITIAL_STATE);

  const getRubikTiles = (faceDirection: FaceDirection): Color[] => {
    const face = cube.find((x) => x.faceDirection === faceDirection);
    return face?.tiles ?? [];
  };

  return (
    <>
      <div data-testid='rubik-cube-container'>
        <div className='ml-[240px]'>
          <Face tiles={getRubikTiles('U')} />
        </div>
        <div className='flex'>
          <Face tiles={getRubikTiles('L')} />
          <Face tiles={getRubikTiles('F')} />
          <Face tiles={getRubikTiles('R')} />
          <Face tiles={getRubikTiles('B')} />
        </div>
        <div className='ml-[240px]'>
          <Face tiles={getRubikTiles('D')} />
        </div>
      </div>

      <Controls {...{ cube, setCube }} />
    </>
  );
};

export default Rubik;
