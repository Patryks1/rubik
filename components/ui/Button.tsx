import React, { FC } from 'react';

type Props = {
  [x: string]: string | Function
};

const Button: FC<Props> = ({ children, ...props }) => (
  <button {...props} className='px-3 py-2 rounded-md bg-slate-600 text-white'>
    {children}
  </button>
);

export default Button;
