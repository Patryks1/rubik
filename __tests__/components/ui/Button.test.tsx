import { render, cleanup, screen } from '@testing-library/react';
import React from 'react';
import { Button } from '@/components/ui';

describe('Button', () => {
  afterAll(() => {
    cleanup();
  });

  it('renders button', () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });

  it('renders children', () => {
    render(<Button>Test</Button>);

    expect(screen.getByText('Test')).toBeTruthy();
  });
});
