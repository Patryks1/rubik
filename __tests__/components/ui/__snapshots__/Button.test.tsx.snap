// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button 1`] = `
<div>
  <button
    class="px-3 py-2 rounded-md bg-slate-600 text-white"
  />
</div>
`;
